fdbserver/CommitProxyServer.actor.cpp

```diff
@@ -907,7 +907,9 @@ EncryptCipherDomainId getEncryptDetailsFromMutationRef(ProxyCommitData* commitDa
 	if (isSystemKey(m.param1)) {
 		// Encryption domain == FDB SystemKeyspace encryption domain
 		domainId = SYSTEM_KEYSPACE_ENCRYPT_DOMAIN_ID;
-	} else if (commitData->tenantMap.empty() || commitData->encryptMode.mode == EncryptionAtRestMode::CLUSTER_AWARE) {
+	} else if (commitData->tenantMap.empty() || commitData->encryptMode.mode == EncryptionAtRestMode::CLUSTER_AWARE ||
+	           (m.type == MutationRef::SetVersionstampedKey && m.param1.size() >= 4 &&
+	            parseVersionstampOffset(m.param1) < 8)) {
```

Comment:
It isn't allowed that a versionstamp mutation modifies the first 8 bytes of a key in required tenant mode.

Reply:
I see so we can always rely on the first 8 bytes then in required mode?

Reply:
This gets called in `assignMutationsToStorageServers` which happens after `applyMetadataToCommittedTransactions` so i think by this point we would have already rejected the transaction. So this check should be able to rely on the fact that in required mode we have the first 8 bytes as valid